### ISSUE TITLE
Compilation issue on ubuntu

### ISSUE THREAD

**Issue by srikanth007m** [yasuoka/mleakdetect#1]
@yasuoka I have tried compiling your tool on ubuntu and ended with below issue. Is this tool developed for BSD Linux?

cc -fPIC -g -Wall -shared -o mleakdetect.so mleakdetect.c
mleakdetect.c:18:10: fatal error: sys/tree.h: No such file or directory
   18 | #include <sys/tree.h>
      |          ^~~~~~~~~~~~
compilation terminated.

**Comment by yasuoka** [comment]
Yes, it's currently for OpenBSD.  I suppose it needs some work for run on Linux.

**Comment by srikanth007m** [comment]
@yasuoka Is it possible to make it work on Linux. Can you help make the changes?

**Comment by yasuoka** [comment]
> @yasuoka Is it possible to make it work on Linux. Can you help make the changes?

I'm not sure for this moment.  I tried that on last weekend.  I made it compiled but it doesn't start with a SEGV.  So it seems I need to study LD_PRELOAD on Linux.
